openpifpaf/annotation.py

```diff
@@ -26,11 +27,13 @@ def __init__(self, keypoints, skeleton, sigmas=None, *,
         self.frontier_order = []
 
         self.skeleton_m1 = (np.asarray(skeleton) - 1).tolist()
-
-        self.score_weights = np.ones((len(keypoints),))
+        if not score_weights:
```

Comment:
I'd do a small adjustment here to make it work for lists and numpy arrays. Check specifically for `is None` here and then below use `np.asarray` instead of `np.array`.